cmake/modules/SwiftUtils.cmake

```diff
@@ -3,3 +3,12 @@ function(precondition var)
     message(FATAL_ERROR "Error! Variable ${var} is false or not set!")
   endif()
 endfunction()
+
+macro(list_replace lst old new)
```

Comment:
We are trying really hard to use functions when possible.  Could you change this into a function?  You can use PARENT_SCOPE to set the variable in the caller.